### PR TITLE
Wallet import: Surface detailed errors

### PR DESCRIPTION
**Pull request by dennisreimann**
- Improves the wallet import and parsers by exposing more concrete error messages.
- Orders existing parsers by specificity, highest to lowest
- Specify the wallet file source when importing

grafik

### CODE REVIEW

File: BTCPayServer/Hosting/BTCPayServerServices.cs [hunk 1]
```diff
@@ -525,11 +525,11 @@ public static void AddOnchainWalletParsers(IServiceCollection services)
...
             services.AddSingleton<IWalletFileParser, NBXDerivGenericWalletFileParser>();
             services.AddSingleton<IWalletFileParser, ElectrumWalletFileParser>();
-            services.AddSingleton<IWalletFileParser, OutputDescriptorWalletFileParser>(provider => provider.GetService<OutputDescriptorWalletFileParser>());
-            services.AddSingleton<OutputDescriptorWalletFileParser>();
             services.AddSingleton<IWalletFileParser, SpecterWalletFileParser>();
-            services.AddSingleton<IWalletFileParser, OutputDescriptorJsonWalletFileParser>();
             services.AddSingleton<IWalletFileParser, WasabiWalletFileParser>();
+            services.AddSingleton<IWalletFileParser, OutputDescriptorWalletFileParser>(provider => provider.GetService<OutputDescriptorWalletFileParser>());
+            services.AddSingleton<IWalletFileParser, OutputDescriptorJsonWalletFileParser>();
+            services.AddSingleton<OutputDescriptorWalletFileParser>();
```
Comment (dennisreimann) [inline BTCPayServer/Hosting/BTCPayServerServices.cs:532]: This reorders them to have the ones with high specificity first.

File: BTCPayServer/Services/WalletFileParsers.cs [hunk 2]
```diff
@@ -31,14 +31,15 @@ public bool TryParseWalletFile(string fileContents, BTCPayNetwork network, [Mayb
...
                     return true;
             }
-            catch (Exception)
+            catch (Exception e)
             {
+                error = e.Message;
             }
         }
-        error = "Unsupported file format";
+        error ??= "Unsupported file format";
```
Comment (dennisreimann) [inline BTCPayServer/Services/WalletFileParsers.cs:42]: Use this as a last resort.

**Comment by Kukks** [comment]
How did we end up with that horrible source dropdown. If this is the result, we should just revert the whole thing.

**Comment by dennisreimann** [comment]
@Kukks No worries, for now it's just an exploration for something @NicolasDorier and I chatted about.

**Comment by Kukks** [comment]
> @Kukks No worries, for now it's just an exploration for something @NicolasDorier and I chatted about.

Kill it with fire and then some more fire and then turn off the fire with a fire hydrant that shoots out oil

**Comment by NicolasDorier** [comment]
How about 

image

@Kukks 

Anyway this isn't for the next release.

**Comment by dennisreimann** [comment]
We discussed this in the deisgn call yesterday and decided to try to come up with another solution for handling the cases, in which it isn't clear what's being imported.

**Pull request closed without merging**